examples/bri-3/src/bri/workgroup/workflows/agents/workflowsStorage.agent.ts

```diff
@@ -0,0 +1,144 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from '../../../../../prisma/prisma.service';
+import { Workstep } from '../../worksteps/models/workstep';
+import { WORKFLOW_NOT_FOUND_ERR_MESSAGE } from '../api/err.messages';
+import { Workflow } from '../models/workflow';
+
+@Injectable()
+export class WorkflowStorageAgent extends PrismaService {
+  async getWorkflowById(id: string): Promise<Workflow> {
+    const workflowModel = await this.workflow.findUnique({
+      where: { id: id },
+      include: { worksteps: true },
+    });
+
+    if (!workflowModel) {
+      throw new NotFoundException(WORKFLOW_NOT_FOUND_ERR_MESSAGE);
+    }
+
+    return new Workflow(
+      workflowModel.id,
+      workflowModel.name,
+      workflowModel.worksteps.map((w) => {
+        return new Workstep(
+          w.id,
+          w.name,
+          w.version,
+          w.status,
+          w.workgroupId,
+          w.securityPolicy,
+          w.privacyPolicy,
+        );
+      }),
+      workflowModel.workgroupId,
+    );
+  }
+
+  async getAllWorkflows(): Promise<Workflow[]> {
+    const workflowModels = await this.workflow.findMany({
+      include: { worksteps: true },
+    });
+    return workflowModels.map((w) => {
+      return new Workflow(
+        w.id,
+        w.name,
+        w.worksteps.map((ws) => {
+          return new Workstep(
+            ws.id,
+            ws.name,
+            ws.version,
+            ws.status,
+            ws.workgroupId,
+            ws.securityPolicy,
+            ws.privacyPolicy,
+          );
+        }),
+        w.workgroupId,
+      );
+    });
+  }
+
+  async createNewWorkflow(workflow: Workflow): Promise<Workflow> {
+    const workstepsData = workflow.worksteps.map((w) => {
```

Comment:
why is this needed? variable name is a little bit too general to understand the reasoning behind this code.

Reply:
rename the workstepsData to workstepIds. This is pulling out  just the ids of the worksteps within workflow because that is all prisma needs to add these relationships to the join table within the DB that reflects the many to many relationship between worksteps and workflow